app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -62,61 +60,126 @@ abstract class Syncer {
     lateinit var context: Context
 
     @Inject
-    lateinit var db: AppDatabase
+    lateinit var serviceRepository: DavServiceRepository
+
+    @Inject
+    lateinit var collectionRepository: DavCollectionRepository
 
     @Inject
     lateinit var logger: Logger
 
+    abstract val serviceType: String
+
+    lateinit var account: Account
+    lateinit var extras: Array<String>
+    lateinit var authority: String
+    lateinit var syncResult: SyncResult
```

Comment:
We should avoid `lateinit var` if possible. Could we initialize them in the constructor?

Reply:
Yes, I did that initially. We need to use a assisted inject and factories, so it became a lot of code. I can see however, that constructor initialization is preferable over lateinit. Will undo :)